### PR TITLE
fix: use 64-bit comparison in AVX2 overflow detection

### PR DESCRIPTION
**Pull request by DeVikingMark**
Replaced _mm256_cmpgt_epi32 with _mm256_cmpgt_epi64 in add_small and sub_tiny functions to handle 64-bit values correctly. 

The previous 32-bit comparison could produce incorrect results when overflow affects the upper 32 bits of 64-bit integers.

This aligns the implementation with the AVX512 version and add_mod function, which already use 64-bit comparisons.

### CODE REVIEW

File: miden-crypto/src/hash/algebraic_sponge/rescue/arch/x86_64_avx2.rs [hunk 1]
```diff
@@ -180,7 +180,7 @@ unsafe fn add_small(
 ) -> (__m256i, __m256i, __m256i) {
     unsafe {
         let res_wrapped_s = map3!(_mm256_add_epi64, x_s, y);
-        let mask = map3!(_mm256_cmpgt_epi32, x_s, res_wrapped_s);
+        let mask = map3!(_mm256_cmpgt_epi64, x_s, res_wrapped_s);
```
Comment (huitseeker) [inline miden-crypto/src/hash/algebraic_sponge/rescue/arch/x86_64_avx2.rs]: The fix is correct, but we might want to add a targeted test for overflow at 32-bit boundaries to prevent regression. The current 5M iteration test should catch most issues, but a specific test with values that trigger overflow in the upper 32 bits could provide extra confidence.

File: miden-crypto/src/hash/algebraic_sponge/rescue/arch/x86_64_avx2.rs [hunk 2]
```diff
@@ -189,9 +189,8 @@ unsafe fn add_small(
 #[inline(always)]
 unsafe fn maybe_adj_sub(res_wrapped_s: __m256i, mask: __m256i) -> __m256i {
     // The subtraction is very unlikely to overflow so we're best off branching.
-    // The even u32s in `mask` are meaningless, so we want to ignore them. `_mm256_testz_pd`
-    // branches depending on the sign bit of double-precision (64-bit) floats. Bit cast `mask` to
-    // floating-point (this is free).
+    // `_mm256_testz_pd` branches depending on the sign bit of double-precision (64-bit) floats.
```
Comment (huitseeker) [inline miden-crypto/src/hash/algebraic_sponge/rescue/arch/x86_64_avx2.rs]: The original comment mentioned that "even u32s in mask are meaningless" and explained why we use `_mm256_testz_pd`. This context might be helpful for future maintenance.

Was the removal intentional? The new comment is cleaner, but the removed explanation provided useful implementation details about why the bitcast works.

**Review by huitseeker — COMMENTED** [review]
On second look, this seems to be fixing a cosmetic problem. Post sign extension it's hard to imagine a valid input on which the 32-bit and the 64 bit comparison would differ.
